frontend/src/utils/utils.js

```diff
@@ -1,34 +1,190 @@
+            'Accept': 'application/json',
+            'Content-Type': 'application/json'},
+        credentials: 'include'
+          };
+        const response = await fetch(`${backendUrlAccess}/calendar/events/${content.id}`,options);
+        if (!response.ok) {
+          throw new Error('Something went wrong!');
+        }
+        refetchEvents();
+        onClose();
+      }
+    catch(error) {
+        console.error(error);
+    }
 }
 
+// API Call for Uploading a .ics File for Parsing
+async function handleICSParsing(formData, onEventsImported) {
+    try{
+        const backendUrlAccess = import.meta.env.VITE_BACKEND_ADDRESS;
+        const options = {
+            method: 'POST',
+            body: formData,
+            credentials: 'include'
+        }
+        const response = await fetch(`${backendUrlAccess}/calendar/import-ics`, options);
+        const data = await response.json();
+        if(response.ok) {
+            onEventsImported(data);
+            alert('Events imported successfully!');
+        } else {
+            alert(`Failed to import events: ${data.error}`);
+        }
+    } catch (error) {
+        console.error('Error uploading file:', error);
+        alert('Failed to upload file. Please try again.');
+    }
+}
 
+// Helper Functions
+// Converts JS Date object (Provide example) into ISO String format (Provide example) -> Specifically extracts the date
 function getDateString(date) {
     return date.toISOString().slice(0, 10);
 }
 
+// Converts JS DateTime object (provide exmaple) into readable date format (Thursday, July 18, 2024)
 function getReadableDateStr(dateTime) {
     return dateTime.toLocaleString('en-US', { weekday: 'long', month: 'long', day: 'numeric', year: 'numeric' });
 }
 
+// Converts JS Date object (Provide example) into ISO String format (Provide example) -> Specifically extracts the time
 function getTimeString(date) {
     const hours = date.getHours();
     const minutes = date.getMinutes();
     return `${hours.toString().padStart(2, '0')}:${minutes.toString().padStart(2, '0')}`;
 }
 
+// Converts JS DateTime object (provide exmaple) into readable time format (9:00 AM)
+// TODO: fix issue with 12th hour (either midnight or noon) not being shown properly
+//       currently shows as 0:00 PM, 0:30 AM, etc.
 function getReadableTimeStr(dateTime) {
     const hours = dateTime.getHours();
     const minutes = dateTime.getMinutes();
     const ampm = hours >= 12 ? 'PM' : 'AM';
-
-    return `${hours % 12}:${minutes.toString().padStart(2, '0')} ${ampm}`;
+    const readableHours = hours % 12 === 0 ? 12 : hours % 12;
+    return `${readableHours % 12}:${minutes.toString().padStart(2, '0')} ${ampm}`;
 }
 
-export {logout, getDateString, getTimeString, getReadableDateStr, getReadableTimeStr};
+export {
+    handleSignUp, handleLogin, handleOnLogout,
+    handleICSParsing,
+    createCalendarEvent, handleEventDelete,
+    getDateString, getTimeString, getReadableDateStr, getReadableTimeStr
```

Comment:
If you want to clean this up further, you can separate the utils file into TimeUtils, LoginUtils, and CalendarUtils.